sklearn/feature_extraction/text.py

```diff
@@ -676,12 +682,8 @@ def __init__(self, input='content', encoding='utf-8',
         if max_df < 0 or min_df < 0:
             raise ValueError("negative value for max_df or min_df")
         self.max_features = max_features
-        if max_features is not None:
-            if (not isinstance(max_features, numbers.Integral) or
-                    max_features <= 0):
-                raise ValueError(
-                    "max_features=%r, neither a positive integer nor None"
-                    % max_features)
+        if max_features is not None and max_features < 0:
```

Comment:
@lesteve 
`None` is the current default. I changed this default to 1.0 which now has the same meaning but to keep code compatible, None is still supported. 

I chose not to check specifically for both floats and ints because I based it on the checks in min_df and max_df which both support float and integer value in the same manner  and they just check for `min_df < 0`. But I could change it if you wish :)